File: spec/python/test_expr_bytes_cmp.py
import unittest

from expr_bytes_cmp import ExprBytesCmp

class TestExprBytesCmp(unittest.TestCase):
    def test_expr_bytes_cmp(self):
        r = ExprBytesCmp.from_file("src/fixed_struct.bin")

        self.assertEqual(r.one, b"P")
        self.assertEqual(r.two, b"ACK")

        self.assertEqual(r.is_eq, True)
        self.assertEqual(r.is_ne, False)
        self.assertEqual(r.is_lt, True)
        self.assertEqual(r.is_gt, False)
        self.assertEqual(r.is_le, True)
        self.assertEqual(r.is_ge, False)
        self.assertEqual(r.is_lt2, False)
        self.assertEqual(r.is_gt2, True)

File: spec/lua/test_expr_bytes_cmp.lua
local luaunit = require("luaunit")

require("expr_bytes_cmp")

TestExprBytesCmp = {}

function TestExprBytesCmp:test_expr_bytes_cmp()
    local r = ExprBytesCmp:from_file("src/fixed_struct.bin")

    luaunit.assertEquals(r.one, "P")
    luaunit.assertEquals(r.two, "ACK")

    luaunit.assertEquals(r.is_eq, true)
    luaunit.assertEquals(r.is_ne, false)
    luaunit.assertEquals(r.is_lt, true)
    luaunit.assertEquals(r.is_gt, false)
    luaunit.assertEquals(r.is_le, true)
    luaunit.assertEquals(r.is_ge, false)
    luaunit.assertEquals(r.is_lt2, false)
    luaunit.assertEquals(r.is_gt2, true)
end
